packages/flutter_tools/lib/src/build_runner/resident_web_runner.dart

```diff
@@ -405,159 +757,56 @@ class ResidentWebRunner extends ResidentRunner {
   }
 
   @override
-  Future<void> debugDumpApp() async {
-    try {
-      await _vmService.callServiceExtension(
-        'ext.flutter.debugDumpApp',
-      );
-    } on vmservice.RPCError {
-      return;
-    }
-  }
-
-  @override
-  Future<void> debugDumpRenderTree() async {
-    try {
-      await _vmService.callServiceExtension(
-        'ext.flutter.debugDumpRenderTree',
-      );
-    } on vmservice.RPCError {
-      return;
-    }
-  }
-
-  @override
-  Future<void> debugDumpLayerTree() async {
-    try {
-      await _vmService.callServiceExtension(
-        'ext.flutter.debugDumpLayerTree',
-      );
-    } on vmservice.RPCError {
-      return;
-    }
-  }
-
-  @override
-  Future<void> debugDumpSemanticsTreeInTraversalOrder() async {
-    try {
-      await _vmService.callServiceExtension(
-          'ext.flutter.debugDumpSemanticsTreeInTraversalOrder');
-    } on vmservice.RPCError {
-      return;
-    }
-  }
-
-  @override
-  Future<void> debugTogglePlatform() async {
-    try {
-      final vmservice.Response response = await _vmService.callServiceExtension(
-          'ext.flutter.platformOverride');
-      final String currentPlatform = response.json['value'];
-      String nextPlatform;
-      switch (currentPlatform) {
-        case 'android':
-          nextPlatform = 'iOS';
-          break;
-        case 'iOS':
-          nextPlatform = 'android';
-          break;
+  Future<int> attach({
+    Completer<DebugConnectionInfo> connectionInfoCompleter,
+    Completer<void> appStartedCompleter,
+  }) async {
+    Uri websocketUri;
+    if (supportsServiceProtocol) {
+      // Cleanup old subscriptions. These will throw if there isn't anything
+      // listening, which is fine because that is what we want to ensure.
+      try {
+        await _vmService.streamCancel('Stdout');
+      } on vmservice.RPCError {
+        // Ignore this specific error.
       }
-      if (nextPlatform == null) {
-        return;
+      try {
+        await _vmService.streamListen('Stdout');
+      } on vmservice.RPCError {
+        // Ignore this specific error.
```

Comment:
nit: I'd prefer `// It is safe to ignore this error because ...`

Reply:
Fixed